build-tools/packages/build-cli/src/commands/list.ts

```diff
@@ -24,33 +34,76 @@ export default class ListCommand extends BaseCommand<typeof ListCommand> {
 	static flags = {
 		releaseGroup: releaseGroupFlag({ required: true }),
 		...filterFlags,
+		feed: Flags.string({
+			description:
+				"Filter the resulting packages to those that should be published to a particular npm feed. Use 'official' for public npm.",
+			options: ["official", "internal", "internal-test"],
+			helpGroup: "PACKAGE FILTER",
+			required: false,
+		}),
 		tarball: Flags.boolean({
 			description:
 				"Return packed tarball names (without extension) instead of package names. @-signs will be removed from the name, and slashes are replaced with dashes.",
 			default: false,
 		}),
 	};
 
-	public async run(): Promise<PnpmListEntry[]> {
+	public async run(): Promise<ListItem[]> {
 		const context = await this.getContext();
 		const releaseGroup = context.repo.releaseGroups.get(this.flags.releaseGroup);
+
 		if (releaseGroup === undefined) {
 			this.error(`Can't find release group: ${this.flags.releaseGroup}`, { exit: 1 });
 		}
 
 		const filterOptions = parsePackageFilterFlags(this.flags);
-		const pnpmListResults = await pnpmList(releaseGroup.repoPath);
+		const pnpmListResults: ListItem[] = await pnpmList(releaseGroup.repoPath);
 		const filtered = filterPackages(pnpmListResults, filterOptions)
 			.reverse()
+			.filter((item) => {
+				const config = context.rootFluidBuildConfig?.policy?.packageNames;
+				if (config === undefined) {
+					this.error(`No fluid-build package name policy config found.`);
```

Comment:
I know we discussed something like this in a PR before but can't recall the details... does this immediately exit? Even if it does, maybe add a `throw new Error()` and a comment to state `this.error()` will exit so the throw is just for clarity when reading the code.

Reply:
Yes, this exits the process. There is another method, errorLog, that logs an error without exiting. You can't add a statement after this because the compiler will complain about unreached code. This is standard throughout all commands, so I'm reluctant to add a comment every time we do this.

Reply:
It definitely can tell this code is not reached, at least in VSCode:

<img width="527" alt="image" src="https://github.com/microsoft/FluidFramework/assets/19589/2cc417d5-6427-4cb1-bca7-fbfff3968dc2">

Reply:
Ok I restarted TS server and now it's also complaining here. Interesting. The return type of `never`, maybe? Anyway, discussion tangent to the PR.